src/mame/capcom/cps2.cpp

```diff
@@ -668,6 +668,7 @@ class cps2_state : public cps_state
 		, m_output(*this, "output")
 		, m_io_in0(*this, "IN0")
 		, m_io_in1(*this, "IN1")
+		, m_dsw{ { *this, "DSWA" }, { *this, "DSWB" }, { *this, "DSWC" } }
```

Comment:
Nitpicking level: very high.

Please use a optional_ioport_array instead.

Reply:
Nitpicking makes for quality. :)

I saw optional_ioport_array and considered it, but it looks like it expects the ports to be differentiated numerically since it uses integer_sequence. The board specifically names the banks DIPSWA, DIPSWB, DIPSWC, so I thought for accuracy's sake, the naming should reflect the board labeling to some extent. The cps1 driver also has them named A/B/C for whatever that is worth.

If you think optional_ioport_array is still the right choice, let me know and I'll go ahead and implement it that way.

Reply:
you can also use letters as in: https://git.redump.net/mame/tree/src/mame/atari/triplhnt.cpp#n52

Reply:
... I really should have realized that would work with %c.

Thanks for everyone's help. Updated and pushed.